src/lib.rs

```diff
@@ -198,7 +223,75 @@ impl Contract {
         notify::notify_mentions(desc.as_str(), id);
     }
 
+    #[payable]
+    pub fn add_proposal(&mut self, body: VersionedProposalBody, labels: HashSet<String>) {
+        near_sdk::log!("add_proposal");
+        let id = self.proposals.len();
+        let author_id = env::predecessor_account_id();
+        let editor_id = author_id.clone();
+        require!(
+            self.is_allowed_to_use_labels(
+                Some(editor_id.clone()),
+                labels.iter().cloned().collect()
+            ),
+            "Cannot use these labels"
+        );
+
+        for label in &labels {
+            let mut other_proposals = self.label_to_proposals.get(label).unwrap_or_default();
+            other_proposals.insert(id);
+            self.label_to_proposals.insert(label, &other_proposals);
+        }
+
+        let mut author_proposals = self.author_proposals.get(&author_id).unwrap_or_else(|| HashSet::new());
+        author_proposals.insert(id);
+        self.author_proposals.insert(&author_id, &author_proposals);
+
+        let proposal = Proposal {
+            id,
+            author_id: author_id.clone(),
+            block_height: 0u64, // TODO
+            snapshot: ProposalSnapshot { editor_id, timestamp: env::block_timestamp(), labels, body: body.clone() },
+            snapshot_history: vec![],
+        };
+
+        proposal::repost::publish_to_socialdb_feed(
+            Self::ext(env::current_account_id())
+            .with_static_gas(env::prepaid_gas().saturating_div(3))
+            .set_block_height_callback(proposal.clone()), proposal.clone());
+
+        let desc = get_proposal_description(body);
+
+        notify::notify_mentions(desc.as_str(), id);
+    }
+
+
+    #[private]
+    pub fn set_block_height_callback(&mut self, old_proposal: Proposal) -> u64 {
```

Comment:
I think we should return `{"proposal_id": "u64-stringified value - near_sdk::json_type::U64"}`